mne/cov.py

```diff
@@ -161,14 +162,15 @@ def copy(self):
         """
         return cp.deepcopy(self)
 
-    def as_diag(self, copy=True):
+    def as_diag(self, copy=None):
```

Comment:
Tried it, no warning. I think it needs to be added to everything that has ``copy=True`` changed to ``copy=None``. In this case ``check_copy_dep`` doesn't throw a warning. When ``copy=False``, it is not needed since the default behavior doesn't change.